Sources/TuistKit/Services/TestService.swift

```diff
@@ -245,19 +245,27 @@ final class TestService { // swiftlint:disable:this type_body_length
         }
 
         if let schemeName {
-            guard let scheme = testableSchemes.first(where: { $0.name == schemeName })
+            guard let scheme = graphTraverser.schemes().first(where: { $0.name == schemeName })
             else {
                 throw TestServiceError.schemeNotFound(
                     scheme: schemeName,
                     existing: testableSchemes.map(\.name)
                 )
             }
 
+            if !testableSchemes.contains(where: { $0.name == scheme.name }) {
+                logger.info("\(scheme.name) has no tests to run, finishing early.")
+                return
+            }
```

Comment:
This is the key part of the PR

Reply:
What if we pass a non-existing scheme? Would we end up here anyway? In that case I think we should fail

Reply:
Passing a non-existing schemes would still fail (see the check at line 248)

Reply:
I find the messaging a bit confusing. We are working with three variables here:

| Scheme existence | Tests > 0 | Tests == 0 due to selection | Message |
| ---- |----- | ---- |---|
|  No | - | - | ??? |
| Yes | Yes | - |  ??? |
| Yes | No | No | ??? |
| Yes | No | Yes | ??? |

I wonder if we should just remove the checks here, and shell out to `xcodebuild`. In the 3 first scenarios from above, I believe users will already get from `xcodebuild` the information that they expect. In the last scenario, things will be a bit tricky because they'll see "no tests run," and they might not know that's due to test selection.

Reply:
In the selective tests scenario, `xcodebuild` returns a failure:
```
The scheme AppScheme cannot be built because it contains no buildable targets.
```

But the extra check is actually not necessary as we already check for test actions right below, so I removed that in the latest commit.